File: scripts/commands/commands/eorzea_database.py
from lib.localize import translate_command
from option.ephemeral import ephemeral


async def create_eorzea_database_command():
    command = {
        "name": "Eorzea Database",
        "name_localizations": {},
        "description": "Search the Eorzea Database",
        "description_localizations": {},
        "type": 1,
        "options": (
            {
                "name": "index",
                "name_localizations": {},
                "description": "The index to search in",
                "description_localizations": {},
                "required": True,
                "choices": (
                    {"name": "Item", "name_localizations": {}, "value": "item"},
                    {"name": "Quest", "name_localizations": {}, "value": "quest"},
                ),
                "type": 3,
            },
            {
                "name": "query",
                "name_localizations": {},
                "description": "The search query",
                "description_localizations": {},
                "required": True,
                "type": 3,
            },
            {
                "name": "language",
                "name_localizations": {},
                "description": "The language to search in",
                "description_localizations": {},
                "choices": (
                    {"name": "English", "name_localizations": {}, "value": "en"},
                    {"name": "Japanese", "name_localizations": {}, "value": "ja"},
                    {"name": "German", "name_localizations": {}, "value": "de"},
                    {"name": "French", "name_localizations": {}, "value": "fr"},
                ),
                "type": 3,
            },
            ephemeral,
        ),
    }

    return translate_command(command)

File: commands/src/commands/EorzeaDatabase.kt
package cloud.drakon.dynamisbot.commands

import cloud.drakon.dynamisbot.buildLocalizationMap
import cloud.drakon.dynamisbot.commandName
import cloud.drakon.dynamisbot.discord.Locale
import cloud.drakon.dynamisbot.discord.interaction.applicationcommand.ApplicationCommand
import cloud.drakon.dynamisbot.discord.interaction.applicationcommand.ApplicationCommandOption
import cloud.drakon.dynamisbot.discord.interaction.applicationcommand.ApplicationCommandOptionChoiceString

private var description = "Search the Eorzea Database"

internal suspend fun eorzeaDatabaseCommand() = ApplicationCommand(
    type = 1,
    name = "Eorzea Database".commandName(),
    nameLocalizations = mapOf(
        Locale.Japanese to "エオルゼアデータベース".commandName(),
        Locale.German to "Eorzea-Datenbank".commandName(),
        Locale.French to "Base de données d'Éorzéa".commandName(),
    ),
    description = description,
    descriptionLocalizations = description.buildLocalizationMap(),
    options = buildList {
        var name = "Index"
        description = "The index to search in"

        add(
            ApplicationCommandOption(
                type = 3,
                name = name.commandName(),
                nameLocalizations = name.buildLocalizationMap(true),
                description = description,
                descriptionLocalizations = description.buildLocalizationMap(),
                required = true,
                choices = buildList {
                    name = "Item"

                    add(
                        ApplicationCommandOptionChoiceString(
                            name = name,
                            nameLocalizations = name.buildLocalizationMap(),
                            value = "Item"
                        )
                    )

                    name = "Quest"

                    add(
                        ApplicationCommandOptionChoiceString(
                            name = name,
                            nameLocalizations = name.buildLocalizationMap(),
                            value = "Quest"
                        )
                    )
                }
            )
        )

        name = "Query"
        description = "The search query"

        add(
            ApplicationCommandOption(
                type = 3,
                name = name.commandName(),
                nameLocalizations = name.buildLocalizationMap(true),
                description = description,
                descriptionLocalizations = description.buildLocalizationMap(),
                required = true,
            )
        )

        name = "Language"
        description = "The language to search in"

        add(
            ApplicationCommandOption(
                type = 3,
                name = name.commandName(),
                nameLocalizations = name.buildLocalizationMap(true),
                description = description,
                descriptionLocalizations = description.buildLocalizationMap(),
                choices = buildList {
                    arrayOf("Japanese", "English", "German", "French").forEach {
                        add(
                            ApplicationCommandOptionChoiceString(
                                name = it,
                                nameLocalizations = it.buildLocalizationMap(),
                                value = it
                            )
                        )
                    }
                }
            )
        )

        add(ephemeralOption())
    }
)
